build/Dockerfile

```diff
@@ -4,170 +4,301 @@
 # syntax=docker.io/docker/dockerfile:1
 
 # This dockerfile contains multiple stages to build three final targets.
-# The file was split in a section for each final target.
 
-# Version config that should be set in the bake file.
+# Version configuration that should be set in the bake file.
 ARG BASE_IMAGE
 ARG RUST_VERSION
 ARG GO_VERSION
 ARG FOUNDRY_NIGHTLY_VERSION
-ARG MACHINE_EMULATOR_VERSION
-ARG ROOTFS_VERSION
-ARG LINUX_VERSION
-ARG LINUX_KERNEL_VERSION
 ARG ROM_VERSION
 ARG ROLLUPS_NODE_VERSION
+ARG MACHINE_EMULATOR_VERSION
+ARG MACHINE_TOOLS_VERSION
+ARG MACHINE_IMAGE_KERNEL_VERSION
+ARG MACHINE_KERNEL_VERSION
+ARG MACHINE_XGENEXT2FS_VERSION
 
 # Build directories.
 ARG SNAPSHOT_BUILD_PATH=/build/snapshot
 ARG DEVNET_BUILD_PATH=/build/devnet
 ARG RUST_BUILD_PATH=/build/rollups-node/rust
 ARG GO_BUILD_PATH=/build/rollups-node/go
 
-# Runtime dir for the cartesi-machine snapshot.
+# Runtime directory for the cartesi-machine snapshot.
 ARG SNAPSHOT_RUNTIME_PATH=/usr/share/cartesi/snapshot
 
-####################################################################################################
-# STAGE: emulator-base
+# =============================================================================
+# STAGE: emulator
+#
+# - Install ca-certificates and curl (setup).
+# - Install the machine-emulator.
+# - Download linux.bin.
+# - Download rootfs.ext2.
 #
-# This stage creates a base-image with the Cartesi machine emulator.
-# The result is used as the base for the snapshot and the node targets.
-# We do this instead of using the cartesi/machine-emulator image to have control over the distro
-# used by the base image.
-FROM ${BASE_IMAGE} as emulator-base
+# NOTE: We do not use the cartesi/machine-emulator image to have control over
+# the distro used by the base image.
+# =============================================================================
+
+FROM ${BASE_IMAGE} as emulator
 
-# Install machine-emulator
 ARG MACHINE_EMULATOR_VERSION
+ARG MACHINE_TOOLS_VERSION
+ARG MACHINE_IMAGE_KERNEL_VERSION
+ARG MACHINE_KERNEL_VERSION
 ARG DEBIAN_FRONTEND=noninteractive
+
+# Install ca-certificates and curl (setup).
+RUN <<EOF
+    set -e
+    apt-get update
+    apt-get install -y --no-install-recommends ca-certificates curl
```

Comment:
```suggestion
    apt-get install -y --no-install-recommends \
        ca-certificates \
        curl
```

Reply:
I think we should avoid breaking lines unless they are longer than the 80 characters limit.
It makes the Dockerfile more readable.
The fact that we are installing ca-certificates and curl is already explained in a commentary.

Reply:
I'd rather have each item in their own lines. IMO, it makes it easier to read and `diff` future additions